content/src/ports/activeEntities.ts

```diff
@@ -0,0 +1,169 @@
+import { getDeployments } from '../service/deployments/deployments'
+import { AppComponents } from '../types'
+
+export type ActiveEntities = {
+  withPointers(...pointers: Pointer[]): Promise<Entity[]>
+  withIds(...entityIds: EntityId[]): Promise<Entity[]>
+  inactivate(...pointers: Pointer[]): void
+  activate(pointer: Pointer, entity: Entity): void
+}
+
+const mapDeploymentsToEntities = (history: PartialDeploymentHistory<Deployment>): Entity[] => {
+  return history.deployments.map(
+    ({ entityVersion, entityId, entityType, pointers, entityTimestamp, content, metadata }) => ({
+      version: entityVersion,
+      id: entityId,
+      type: entityType,
+      pointers,
+      timestamp: entityTimestamp,
+      content: content?.map(({ key, hash }) => ({ file: key, hash })),
+      metadata
+    })
+  )
+}
+
+export const createActiveEntitiesComponent = (
+  components: Pick<AppComponents, 'database' | 'env' | 'logs' | 'metrics'>
+): ActiveEntities => {
+  const logger = components.logs.getLogger('ActiveEntities')
+  const cache = new LRU<EntityId, Entity>({
+    max: components.env.getConfig(EnvironmentConfig.ENTITIES_CACHE_SIZE)
+  })
+  const entityIdByPointers = new Map<Pointer, EntityId>()
+
+  const reportCacheAccess = (entity: Entity, result: 'hit' | 'miss') => {
+    components.metrics.increment('dcl_entities_cache_accesses_total', {
+      entity_type: entity.type,
+      result: result
+    })
+  }
+
+  const cacheEntity = (entity: Entity): void => {
+    cache.set(entity.id, entity)
+  }
+
+  const invalidateEntity = (entityId: EntityId): void => {
+    const entity = cache.get(entityId)
+    if (entity) {
+      for (const pointer of entity.pointers) {
+        entityIdByPointers.delete(pointer)
+      }
+    }
+    cache.del(entityId)
+  }
+
+  const updateCacheAndReturnAsEntities = (deployments: PartialDeploymentHistory<Deployment>): Entity[] => {
+    const entities = mapDeploymentsToEntities(deployments)
+    // Save the calculated values
+    for (const entity of entities) {
+      cacheEntity(entity)
+      for (const pointer of entity.pointers) {
+        entityIdByPointers.set(pointer, entity.id)
+      }
+    }
+    return entities
+  }
+
+  const findEntitiesByIds = async (entityIds: EntityId[]): Promise<Entity[]> => {
+    const deployments = await getDeployments(components, {
+      filters: { entityIds, onlyCurrentlyPointed: true }
+    })
+    return updateCacheAndReturnAsEntities(deployments)
+  }
+
+  const findEntitiesByPointers = async (pointers: Pointer[]): Promise<Entity[]> => {
+    const deployments = await getDeployments(components, {
+      filters: { pointers, onlyCurrentlyPointed: true }
+    })
+    for (const deployment of deployments.deployments) {
+      deployment.pointers.forEach((pointer) => entityIdByPointers.set(pointer, deployment.entityId))
+    }
+    return updateCacheAndReturnAsEntities(deployments)
+  }
+
+  const getByIds = async (...entityIds: EntityId[]): Promise<Entity[]> => {
```

Comment:
`findEntitiesByIds` goes to the DB to retrieve the deployments, where `getByIds` is checking the cache otherwise calling to `findEntitiesByIds` 

added some comments btw